config/initializers/2_limited_federation_mode.rb

```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+Rails.application.configure do
+  config.x.limited_federation_mode = (ENV['LIMITED_FEDERATION_MODE'] || ENV['WHITELIST_MODE']) == 'true'
```

Comment:
@mjankowski I assume this is what's tripping up `config_for`? tbh, `WHITELIST_MODE` has been gone for a long time, I think it's time to remove it, or at least make it stop the server from starting up.

Reply:
Yeah - there are various areas that are generally in the "lets shove this config that comes from an ENV var into a place where it's loaded as early as possible in a way that won't change during running" ... and the `config_for` feature has been the path for a lot of that.

I'm mainly holding off because I have ~5 more open PRs in that area and want those to settle in a bit before adding to the pile. This one should be easy when I get to it.

In this case I noticed the deprecated/renamed thing and figured maybe we can remove the older value as well.